src/printer.js

```diff
@@ -53,23 +53,25 @@ function genericPrint(path, options, printPath, args) {
   assert.ok(path instanceof FastPath);
 
   var node = path.getValue();
-  var parts = [];
-  var needsParens = false;
-  var linesWithoutParens = genericPrintNoParens(path, options, printPath, args);
-
-  if (!node || isEmpty(linesWithoutParens)) {
-    return linesWithoutParens;
-  }
 
   // Escape hatch
```

Comment:
The diff doesn't make this justice, I just moved this block above, I didn't delete and re-add all those lines.